androidplot-core/src/test/java/com/androidplot/util/FastNumberTest.java

```diff
@@ -0,0 +1,172 @@
+
+        assertTrue("Assert 2: NaN should not be == Nan", Double.NaN != Double.NaN);
+        assertTrue("Assert 3: NaN should not be == Nan", new FastNumber(new Double(Double.NaN))
+                .equals(new FastNumber(new Double(Double.NaN))));
+        assertTrue("Assert 4: -0d should be == 0d", 0d == -0d);
+        assertTrue("Assert 5: -0d should not be equals() 0d", !new FastNumber(new Double(0d))
+                .equals(new FastNumber(new Double(-0d))));
+
+        FastNumber dmax = new FastNumber(new Double(Double.MAX_VALUE));
+        FastNumber dmax1 = new FastNumber(new Double(Double.MAX_VALUE));
+
+        assertTrue("Equality test failed", dmax.equals(dmax1) && !(dmax.equals(new Object())));
+    }
+
+    @Test
+    public void testEqualsFloat() {
+        FastNumber f1 = new FastNumber(new Float(8765.4321f));
+        FastNumber f2 = new FastNumber(new Float(8765.4321f));
+        FastNumber f3 = new FastNumber(new Float(-1.0f));
+        assertTrue("Assert 0: Equality test failed", f1.equals(f2) && !(f1.equals(f3)));
+
+        assertTrue("Assert 1: NaN should not be == Nan", Float.NaN != Float.NaN);
+        assertTrue("Assert 2: NaN should not be == Nan", new FastNumber(new Float(Float.NaN)).equals(new FastNumber(new Float(
+                Float.NaN))));
+        assertTrue("Assert 3: -0f should be == 0f", 0f == -0f);
+        assertTrue("Assert 4: -0f should not be equals() 0f", !new FastNumber(new Float(0f)).equals(new FastNumber(new Float(
+                -0f))));
+
+        f1 = new FastNumber(new Float(1098.576f));
+        f2 = new FastNumber(new Float(1098.576f));
+        f3 = new FastNumber(new Float(1.0f));
+        assertTrue("Equality test failed", f1.equals(f2) && !(f1.equals(f3)));
+
+        assertTrue("NaN should not be == Nan", Float.NaN != Float.NaN);
+        assertTrue("NaN should not be == Nan", new FastNumber(new Float(Float.NaN))
+                .equals(new FastNumber(new Float(Float.NaN))));
+        assertTrue("-0f should be == 0f", 0f == -0f);
+        assertTrue("-0f should not be equals() 0f", !new FastNumber(new Float(0f)).equals(new FastNumber(new Float(-0f))));
+    }
+
+    @Test
+    public void testHashCodeInteger() {
+        assertEquals(1, new FastNumber(new Integer(1)).hashCode());
+        assertEquals(2, new FastNumber(new Integer(2)).hashCode());
+        assertEquals(0, new FastNumber(new Integer(0)).hashCode());
+        assertEquals(-1, new FastNumber(new Integer(-1)).hashCode());
+    }
+
+    @Test
+    public void testHashCodeInteger2() {
+        // Test for method int java.lang.Integer.hashCode()
+
+        FastNumber i1 = new FastNumber(new Integer(1000));
+        FastNumber i2 = new FastNumber(new Integer(-1000));
+        assertTrue("Returned incorrect hashcode", i1.hashCode() == 1000
+                && (i2.hashCode() == -1000));
+    }
+
+    @SuppressWarnings("PointlessBitwiseExpression")
+    @Test
+    public void testHashCodeLong() {
```

Comment:
I'm not sure theres much value in verifying that the hashcode method generates an expected value for a subset of specific inputs and It's probably unrealistic to reasonably test `FastNumber.hashcode` thoroughly.

Also, all we really need validate is that FastNumber calls through to the appropriate `Number` implementation's hashCode method.

My suggestion would be to make one test (instead of separate ones for `Long`, `Integer` and `Double`) and do some permutations of asserting the hashcode of various values are equal or unequal.